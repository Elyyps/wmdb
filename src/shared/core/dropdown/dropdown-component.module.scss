.dropdown {
  width: 100%;
}

.form {
  > * {
    margin-top: 24px;
    @media (max-width: $breakpoint-xsmall-max) {
      margin-top: 8px;
    }
  }

  .uk-margin-small-top {
    margin-top: 8px;
  }

  &__list {
    display: flex;
    align-items: center;
    margin-bottom: 23px;
    > * {
      margin-right: 23px;
    }
  }

  &__item {
    &:not(:first-child) {
      margin-top: 24px;
    }
    height: 45px;
    background: $global-background;
    border: 1px solid $gray-lighter;
    box-shadow: inset 0 1px 2px 0 rgba($black, 0.2);

    padding-right: 8px;
    box-sizing: border-box;
    border-radius: 4px;
    border: 1px solid $gray-lighter;
    font-family: $global-font-family;
    box-shadow: inset 0 1px 2px 0 rgba($black, 0.2);
    &:focus {
      outline: none;
      border-color: $global-color;
    }
    select {
      font-size: 16px;
      font-family: $global-font-family;
      margin-right: 40px !important;
      height: 43px;
      border: none;
      background: url(data:image/svg+xml;base64,PHN2ZyBpZD0iTGF5ZXJfMSIgZGF0YS1uYW1lPSJMYXllciAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA0Ljk1IDEwIj48ZGVmcz48c3R5bGU+LmNscy0ye2ZpbGw6IzQ0NDt9PC9zdHlsZT48L2RlZnM+PHRpdGxlPmFycm93czwvdGl0bGU+PHBvbHlnb24gY2xhc3M9ImNscy0yIiBwb2ludHM9IjEuNDEgNC42NyAyLjQ4IDMuMTggMy41NCA0LjY3IDEuNDEgNC42NyIvPjxwb2x5Z29uIGNsYXNzPSJjbHMtMiIgcG9pbnRzPSIzLjU0IDUuMzMgMi40OCA2LjgyIDEuNDEgNS4zMyAzLjU0IDUuMzMiLz48L3N2Zz4=)
        no-repeat 100% 50%;
      background-size: 32px 32px;
      
      appearance: none;

      &:focus {
        outline: 0;
      }
    }
    select::-ms-expand {
      display: none;
  }
    &.isIcon {
      input {
        padding-left: 48px;
      }
      .form__item-holder {
        > .icon-svg {
          position: absolute;
          left: 16px;
          top: 50%;
          transform: translateY(-50%);
        }
      }
    }
    &.error {
      .input,
      .textarea {
        border-color: $global-danger-background;
      }

      .input-icon {
        .icon-svg {
          svg,
          path {
            stroke: $global-danger-background;
          }
        }
      }
      .error-message {
        font-size: 14px;
        color: $global-danger-background;
      }
    }
    &.success {
      .form__icon {
        .icon-svg {
          svg,
          path {
            stroke: #00a82f;
            fill: none;
          }
        }
      }
    }
    &.search {
      .icon-svg {
        position: absolute;
        left: 16px;
        top: 50%;
        transform: translateY(-50%);
      }
    }
  }

  &__item-holder {
    position: relative;
    .icon-svg {
      path {
        fill: transparent;
      }
    }
  }

  &__item-label {
    font-size: 16px;
    font-weight: 600;
    margin-bottom: 8px;
    color: $global-color;
    display: block;
  }

  &__item-icon {
    cursor: pointer;
    position: absolute;
    right: 12px;
    top: 50%;
    transform: translateY(-50%);

    .icon-svg {
      svg,
      path {
        fill: currentColor;
      }
    }
  }
}
