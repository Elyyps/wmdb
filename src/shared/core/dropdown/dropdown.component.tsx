import React, { useState } from "react";
import classNames from "classnames";
import styles from "./dropdown-component.module.scss";
import { IDropdownItem } from "../../api/core/dropdown-item";

interface IDropdownProps {
  classModify?: string;
  errorMessage?: string;
  isError?: string;
  isSuccess?: boolean;
  label?: string;
  name?: string;
  onChange?: (value: any) => void;
  options?: IDropdownItem[];
  placeholder?: string;
}

const Dropdown = (props: IDropdownProps) => {
  const { name, options, placeholder, classModify, label, isError, isSuccess } = props;
  const selectClassName = classNames(styles["dropdown"], {
    [styles[`dropdown--${classModify}`]]: classModify
  });

  const wrapperClassnames = classNames({
    ["error"]: isError,
    ["success"]: isSuccess
  });

  const [value, setValue] = useState("");

  const handleChange = (event: any) => {
    {
      setValue(event.target.value);
      if (props.onChange) {
        props.onChange(event.target.value);
      }
    }
  };

  return (
    <React.Fragment>
      <div className={`${styles["form__item"]} ${wrapperClassnames}`}>
        {label && (
          <label htmlFor="label" className={styles["form__item-label"]}>
            {label}
          </label>
        )}
        <div className={styles["form__item-holder"]}>
          <select className={selectClassName} name={name} onChange={handleChange} value={value ? value : "all"}>
            {placeholder && (
              <option value="all" disabled aria-selected>
                {placeholder}
              </option>
            )}

            {options &&
              options.map((option, key) => (
                <option aria-selected key={key} value={option.value}>
                  {option.text}
                </option>
              ))}
          </select>
        </div>

        {isError && <div className={styles["error-message"]}>{isError && isError}</div>}
      </div>
    </React.Fragment>
  );
};

export { Dropdown };
