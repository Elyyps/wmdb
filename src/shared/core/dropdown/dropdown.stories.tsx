import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { Dropdown } from "./dropdown.component";

storiesOf("Dropdown", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <Dropdown
      options={[
        { label: "Label item #1", value: "value item #1" },
        { label: "Label item #2", value: "value item #2" },
        { label: "Label item #3", value: "value item #3" },
        { label: "Label item #4", value: "value item #4" },
        { label: "Label item #5", value: "value item #5" },
        { label: "Label item #6", value: "value item #6" }
      ]}
      placeholder="placeholder"
    />
  ));
