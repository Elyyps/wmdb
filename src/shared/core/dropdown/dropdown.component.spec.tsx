import React from "react";
import { shallow } from "enzyme";
import { Dropdown } from "./dropdown.component";

describe("[Dropdown]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<Dropdown />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
