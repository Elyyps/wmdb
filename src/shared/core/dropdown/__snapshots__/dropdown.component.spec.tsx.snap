// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Dropdown] should render component without crashing 1`] = `
<Fragment>
  <div
    className="form__item "
  >
    <div
      className="form__item-holder"
    >
      <select
        className="dropdown"
        onChange={[Function]}
        value={-1}
      />
    </div>
  </div>
</Fragment>
`;
